$OpenBSD$

Unify the HTTP2 client changes for http2-5.x and local SocksCredentials handling.

--- src/Simplex/Messaging/Transport/HTTP2/Client.hs.orig
+++ src/Simplex/Messaging/Transport/HTTP2/Client.hs
@@ -26,6 +26,7 @@
 import Control.Concurrent.Async
 import Control.Exception (IOException, try)
 import qualified Control.Exception as E
+import qualified Data.ByteString.Char8 as B
 import Control.Monad
 import Data.Functor (($>))
 import Data.Time (UTCTime, getCurrentTime)
@@ -35,14 +36,13 @@
 import Network.HTTP2.Client (ClientConfig (..), Request, Response)
 import qualified Network.HTTP2.Client as H
 import Network.Socket (HostName, ServiceName)
-import Network.Socks5 (SocksCredentials)
 import qualified Network.TLS as T
 import Numeric.Natural (Natural)
 import qualified Simplex.Messaging.Crypto as C
 import Simplex.Messaging.Encoding.String
 import Simplex.Messaging.Protocol (NetworkError (..), toNetworkError)
 import Simplex.Messaging.Transport (ALPN, STransportPeer (..), SessionId, TLS (tlsALPN, tlsPeerCert, tlsUniq), TransportPeer (..), TransportPeerI (..), getServerVerifyKey)
-import Simplex.Messaging.Transport.Client (TransportClientConfig (..), TransportHost (..), defaultTcpConnectTimeout, runTLSTransportClient)
+import Simplex.Messaging.Transport.Client (SocksCredentials (..), TransportClientConfig (..), TransportHost (..), defaultTcpConnectTimeout, runTLSTransportClient)
 import Simplex.Messaging.Transport.HTTP2
 import Simplex.Messaging.Util (eitherToMaybe)
 import UnliftIO.STM
@@ -213,4 +213,4 @@
 runHTTP2ClientWith bufferSize host setup client = setup $ \tls -> withHTTP2 bufferSize (run tls) (pure ()) tls
   where
     run :: TLS p -> H.Config -> IO a
-    run tls cfg = H.run (ClientConfig "https" (strEncode host) 20) cfg $ client tls
+    run tls cfg = H.run (H.defaultClientConfig { H.scheme = "https", H.authority = B.unpack (strEncode host), H.cacheLimit = 20 }) cfg $ client tls
